Upgrade osmdroid

francoisfds opened the pull request:
Hello.

It seems that Mapnik tiles is suffering download errors on recent Android versions (due to limitations raised by the OpenStreetMap administrators).
For example an Android 10 device face this issue (whereas Android 5 don't have it).

This PR do the following :
  - satisfy the new OSM requirements in order to solve #51
  - get osmdroid up to date
  - solves #34 as side effect (new osmdroid improves zoom quality).

Changes have been tested on Android 10 : working ; Android 5 is still working.

bparmentier commented on app/src/main/java/be/brunoparmentier/openbikesharing/app/activities/MapActivity.java:
```diff
@@ -96,6 +99,16 @@ protected void onCreate(Bundle savedInstanceState) {
+        } catch (PackageManager.NameNotFoundException e) {
+            e.printStackTrace();
+        }
+        Configuration.getInstance().setUserAgentValue(context.getPackageName() + "/" + versionCode);
```
I don't think we really need the versionCode in the UserAgent + I'd use `BuildConfig.APPLICATION_ID` which identifies the app, instead of `context.getPackageName()` (the package name can be different from the application ID).

So I'd replace the whole block with:
```
Configuration.getInstance().setUserAgentValue(BuildConfig.APPLICATION_ID)
```
(as suggested in https://github.com/osmdroid/osmdroid/wiki/Important-notes-on-using-osmdroid-in-your-app#set-the-http-user-agent-variable)

bparmentier commented on app/src/main/java/be/brunoparmentier/openbikesharing/app/activities/MapActivity.java:
```diff
@@ -96,6 +99,16 @@ protected void onCreate(Bundle savedInstanceState) {
+            e.printStackTrace();
+        }
+        Configuration.getInstance().setUserAgentValue(context.getPackageName() + "/" + versionCode);
+
```
I don't think we really need the versionCode in the UserAgent + I'd use `BuildConfig.APPLICATION_ID` which identifies the app, instead of `context.getPackageName()` (the package name can be different from the application ID).

So I'd replace the whole block with:
```
Configuration.getInstance().setUserAgentValue(BuildConfig.APPLICATION_ID)
```
(as suggested in https://github.com/osmdroid/osmdroid/wiki/Important-notes-on-using-osmdroid-in-your-app#set-the-http-user-agent-variable)

francoisfds replied:
Hi, yeah it's true that the osmdroid does not officially recommend to go as far as this. Actually, as stated in my commit message I relied on the osmdroid github thread that pointed the issue, and more precisely a comment of an OSM sysadmin called Firefishy.
My understanding of the whole bug is that, at some point of the time when there was a single user-agent for every osmdroid client, it has been banned by the OSM Foundation because of overloading problems in which it was involved. Now the new osmdroid requires the customization of the user-agent so OSM could ban only the ending client that overloads its servers. In this context using the APPLICATION_ID would be enough. That said, the recommendation of Firefishy was to use a different user-agent for each app version to prevent the whole package to be banned forever if a specific version behaves badly for some reason.

bparmentier commented on app/src/main/java/be/brunoparmentier/openbikesharing/app/activities/StationActivity.java:
```diff
@@ -93,7 +106,7 @@ protected void onCreate(Bundle savedInstanceState) {
                 break;
             case MAP_LAYER_CYCLEMAP:
-                map.setTileSource(TileSourceFactory.CYCLEMAP);
+                map.setTileSource(new ThunderforestTileSource(context, ThunderforestTileSource.CYCLE));
```
Doesn't this source require an API key?

francoisfds replied:
Indeed. I just made this change because the old code was not compiling, but it would be better to use a legit source instead.

francoisfds replied:
I should have remove it. I can re-push with the removing unless you think about another source to add...

francoisfds replied:
Finally re-pushed with another source (HikeBikeMap). Looking for your review. Regards.

francoisfds committed:
Update osmdroid to v6.1.8

This commit aims to get Mapnik working for newer Android versions,
which is done by upgrading osmdroid to latest. This update implies
some porting work :

- use Double instead of int argument when calling setMinZoomLevel.

- enhance osmdroid configuration : call 'load' to ensure config
(github.com/osmdroid/osmdroid/wiki/How-to-use-the-osmdroid-library)
and customize user-agent to access tiles following recommendations
(github.com/osmdroid/osmdroid/wiki/Important-notes-on-using-osmdroid-
in-your-app#set-the-http-user-agent-variable).

- replace the discarded OpenCycleMap tile source by HikeBikeMap.

francoisfds commented:
Hello ! Please find a new PR including your suggestions and the new osmdroid updated in the meantime. Regards.

ildar commented:
then close this PR ( in favor of which? )

francoisfds commented:
Hi everyone. Sorry I missworded, actually I overwrited with a new commit, not a new PR 😛
See the commit tab, the current one (e1af67f) is up to date with the last demands of @bparmentier, and ready for merge if that's OK ?

Please let me know of any feedback about this. 
Regards.
